testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/ejb/remote/EJBOverHTTPTestCase.java

```diff
@@ -0,0 +1,619 @@
+        waitForProxyRegistration();
+
+        // check multiple SFSB sessions are created on different nodes
+        testMultipleSFSBWithoutFailover(() -> new RemoteEJBDirectory(MODULE_NAME, getProperties(false)));
+    }
+
+    @Ignore
+    @InSequence(6)
+    @Test
+    public void testSFSBInTransactionWithoutFailover() throws Exception {
+        waitForProxyRegistration();
+
+        // check SFSB failover for a single SFSB
+        testSFSBInTransactionWithoutFailover(() -> new RemoteEJBDirectory(MODULE_NAME, getProperties(false)));
+    }
+
+    /*
+     * A test which checks SLSB behaviour of EJB client with EJB/HTTP in the absence of failover.
+     * The key behaviour to validate: load balancing.
+     * Proxies are obtained via JNDI/HTTP and invocations are made using EJB/HTTP.
+     */
+    public void testSLSBWithoutFailover(ExceptionSupplier<EJBDirectory, Exception> directoryProvider) throws Exception {
+
+        log.infof(MODULE_NAME+ " : testSLSBWithoutFailover: starting test");
+        try (EJBDirectory directory = directoryProvider.get()) {
+            Incrementor slsb = directory.lookupStateless(StatelessIncrementorBean.class, Incrementor.class);
+            logAffinityForBean(slsb, "testSLSBWithoutFailover");
+
+            Result<Integer> result = slsb.increment();
+            log.info("Called SLSBWithoutFailover: SLSB backend node = " + result.getNode());
+
+            int count = 1;
+            for (int i = 0; i < COUNT; ++i) {
+                result = slsb.increment();
+                log.info("Called SLSBWithoutFailover: SLSB backend node = " + result.getNode());
+            }
+        }
+    }
+
+    /*
+     * A test which checks SFSB behaviour of EJB client with EJB/HTTP in the absence of failover.
+     * The key behaviour to validate: stickiness of EJB sessions to the nodes that own them.
+     * Proxies are obtained via JNDI/HTTP and invocations are made using EJB/HTTP.
+     */
+    public void testSFSBWithoutFailover(ExceptionSupplier<EJBDirectory, Exception> directoryProvider) throws Exception {
+
+        log.infof(MODULE_NAME+ " : testSFSBWithoutFailover: starting test");
+        try (EJBDirectory directory = directoryProvider.get()) {
+            // this single statement creates a session on the server
+            Incrementor sfsb = directory.lookupStateful(StatefulIncrementorBean.class, Incrementor.class);
+            logAffinityForBean(sfsb, "testSFSBWithoutFailover");
+
+            Result<Integer> result = sfsb.increment();
+            log.infof("Called SFSBWithoutFailover: SFSB value = %s, backend node = %s", result.getValue().intValue(), result.getNode());
+
+            int count = 1;
+            for (int i = 0; i < COUNT; ++i) {
+                result = sfsb.increment();
+                log.infof("Called SFSBWithoutFailover: SFSB value = %s, backend node = %s", result.getValue().intValue(), result.getNode());
+            }
```

Comment:
@rachmatowicz I guess, since we have SFSB, that sticky session applies here; is that the expected behavior?